h/auth/util.py

```diff
@@ -14,17 +11,10 @@ def client_authority(request):
     """
     Return the authority associated with an authenticated auth_client or None.
 
-    Once a request with an auth_client is authenticated, a principal is set
-    indicating the auth_client's verified authority
-
-    see :func:`~h.auth.util.principals_for_auth_client` for more details on
-    principals applied when auth_clients are authenticated
-
     :rtype: str or None
     """
-    for principal in request.effective_principals:
-        match = re.match(r"^client_authority:(.+)$", principal)
-        if match and match.group(1):
-            return match.group(1)
+    # This function is kind of dumb and should be removed...
+    if request.identity and request.identity.auth_client:
+        return request.identity.auth_client.authority
```

Comment:
I think this is kind of showing this is silly. Hopefully this will end up just boiling away to nothing as the refactors continue. I can't see this surviving the move to non ACL based authorization.

Reply:
Ok, so previously this looked for a `client_authority:<AUTHORITY>` principal (if any) in `request.effective_prinicipals` and returned the `<AUTHORITY>` part. There aren't going to be any principals anymore in Pyramid 2.0, there's a custom `request.identity` object instead, so the new version just returns `request.identity.auth_client.authority` (or `None`)

Reply:
Yes, but it's a bit more circular. Previously:

* We generated all principals for based on your user and `auth_client`
* If you were an `auth_client` one of them would be `client_authority:<auth_client_authority>`
* Then we searched through your principals for a matching string
* Parsed out the `auth_client_authority`

Now:

* The identity contains the `auth_client` if you have one
* We return the `auth_client_authority` from it

The object we use for an identity is custom, but the presence of `request.identity` isn't. I've added it as a request method for now to smooth our transition from 1.x to 2.0. When this is done it'll be removed as Pyramid provides it.